openedx/adg/lms/applications/handlers.py

```diff
@@ -0,0 +1,31 @@
+"""
+handler methods for applications
+"""
+
+from django.contrib.auth.models import Group
+from django.db.models.signals import post_delete, pre_save
+from django.dispatch import receiver
+
+from .models import BusinessLine
+
+
+@receiver(pre_save, sender=BusinessLine)
+def update_business_line_group(instance, **kwargs):  # pylint: disable=unused-argument
+    """
+    Add user group on creating a Business Line
+    Modify user group on updating a Business Line
+    """
+    if instance.pk is None:
+        Group.objects.create(name=instance.title)
+    else:
+        old_instance = BusinessLine.objects.get(pk=instance.pk)
+        if old_instance.title != instance.title:
+            Group.objects.filter(name=old_instance.title).update(name=instance.title)
```

Comment:
No, instance.title will have updated record. For example, if 'foo' is added to the Business Line and then changed to 'foobar', update_or_create will just add a new Business Line instead of updating the existing one.